Runtime/Highlight/DefaultHighlighter.cs

```diff
@@ -0,0 +1,445 @@
+            }
+        }
+
+        protected void ReenableRenderers()
+        {
+            if (highlightMeshRenderer != null)
+            {
+                highlightMeshRenderer.enabled = false;
+                highlightMeshRenderer.gameObject.SetActive(false);
+            }
+
+            foreach (Renderer renderer in renderers)
+            {
+                if (renderer != null)
+                {
+                    renderer.enabled = true;
+                }
+            }
+        }
+
+        protected Material NewHighlightMaterial(Color highlightColor)
+        {
+            Material material = CreateHighlightMaterial();
+            material.color = highlightColor;
+
+            // In case the color has some level of transparency,
+            // we set the Material's Rendering Mode to Transparent. 
+            if (Mathf.Approximately(highlightColor.a, 1f) == false)
+            {
+                material.SetInt("_SrcBlend", (int)BlendMode.One);
+                material.SetInt("_DstBlend", (int)BlendMode.OneMinusSrcAlpha);
+                material.SetInt("_ZWrite", 0);
+                material.DisableKeyword("_ALPHATEST_ON");
+                material.DisableKeyword("_ALPHABLEND_ON");
+                material.EnableKeyword("_ALPHAPREMULTIPLY_ON");
+                material.renderQueue = 3000;
+            }
+            
+            return material;
+        }
+        
+        protected Material NewHighlightMaterial(Texture mainTexture)
+        {
+            Material material = CreateHighlightMaterial();
+            material.mainTexture = mainTexture;
+            return material;
+        }
+
+        protected Material CreateHighlightMaterial()
+        {
+            string shaderName = GraphicsSettings.currentRenderPipeline ? "Universal Render Pipeline/Lit" : "Standard";
+            Shader defaultShader = Shader.Find(shaderName);
+
+            if (defaultShader == null)
+            {
+                throw new NullReferenceException($"{name} failed to create a default material," + 
+                    $" shader \"{shaderName}\" was not found. Make sure the shader is included into the game build.");
+            }
+
+            return new Material(defaultShader);
+        }
+
+        protected bool CanObjectBeHighlighted()
+        {
+            if (enabled == false)
+            {
+                Debug.LogError($"{GetType().Name} component is disabled for {name} and can not be highlighted.", gameObject);
+                return false;
+            }
+            
+            if (gameObject.activeInHierarchy == false)
+            {
+                Debug.LogError($"{name} is disabled and can not be highlighted.", gameObject);
+                return false;
+            }
+
+            return true;
+        }
+        
+        private bool IsTheLastKey(string highlightID)
+        {
+            return activeHighlights.ContainsKey(highlightID) && activeHighlights.Last().Key == highlightID;
```

Comment:
first check is irrelevant, also doing a last lookup which is a extension method every update is kinda expensive... i guess?

Reply:
True, once the dictionary is made into a list I could get rid off linq